src/stac_api_validator/__main__.py

```diff
@@ -64,6 +64,10 @@
     "--auth-query-parameter",
     help="Query parameter key and value to pass for authorization, e.g., 'key=xyz'.",
 )
+@click.option(
+    "--auth-headers",
+    help="Auth headers to attach to the request, e.g., {'Authorization': 'api-key <api-key>'}",
```

Comment:
Making this argument literally Python code is a little strange to me. I'd recommend using the same format as curl, e.g. (from their docs):

```
curl -H "X-First-Name: Joe" https://example.com
```

So, for your example, the expected string would be `"Authorization: api-key <api-key>"`.

Reply:
I'd prefer for this to be `--auth-header` and a click "multiple" field, where each header specified is the HTTP format style that curl uses (as Pete said)